admin/scripts/zvlastni/reporty/maily-dle-data-ucasti.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+require_once('sdilene-hlavicky.php');
+
+isset($_GET['start']) ? $start = $_GET['start'] : $start = 0; //nastavení startu pro filtr v SQL dotazu - čím vyšší číslo, tím déle účastník nebyl na GC
+
+$poleUcastniku[] = null;
+  $o = dbQuery('
+    SELECT u.email1_uzivatele
+    FROM uzivatele_hodnoty u
+    LEFT JOIN r_uzivatele_zidle uz ON uz.id_uzivatele = u.id_uzivatele AND uz.id_zidle MOD 100 = -1
+    WHERE u.nechce_maily IS NULL
+    GROUP BY u.id_uzivatele
+    ORDER BY MIN(COALESCE(uz.id_zidle, 0))
+    LIMIT '.$start.','.($start+2000) // filtruje účastníky vždy po 2000 (kvůli omezení mailchimpu)
+  );
+while($r = mysqli_fetch_assoc($o)) {
+  if (strpos($r["email1_uzivatele"],'@')) {
```

Comment:
Filtraci dej primo do SQL a nasledne pouzij zSql jak bylo puvodne, vyrazne se to zkrati. Viz puvodni kod. Jak to budes mit, tak si precti cely kod znova. Jestli ti tam pak nezustaly treba nejake nepouzite promenne.